### PR TITLE
Feat: add GitHub actions 

### LINKED ISSUE #6: add GitHub actions
Add GitHub Actions for the project which runs everything inside the `.ci/` folder and makes sure it passes on every commit.

Refer to the [documentation](https://docs.github.com/en/actions) for in-depth tutorial of GitHub Actions.

### PR DESCRIPTION
**Pull request by pearguacamole**
Closes #6 
---

### Pull Request Description

- **Feature Addition**: 
  - Implemented a GitHub Actions workflow that automatically runs `.ci/test.sh`, `.ci/lint-plugin.sh` and `.ci/lint-shell.sh`after every commit, ensuring consistent testing and feedback.

- **Fix for Exit Codes**: 
  - Updated the `.ci/` scripts to ensure they return the correct exit codes, allowing for proper signaling of success or failure in the CI pipeline.

### Issue Addressed
This PR resolves the issue **Add GitHub Actions for the project** by implementing a CI workflow that runs all necessary scripts located in the `.ci/` folder and ensures they pass on every commit. 

For more details on how to utilize GitHub Actions, please refer to the [GitHub Actions Documentation](https://docs.github.com/en/actions).

---

### CODE REVIEW

File: .ci/lint-plugin.sh [hunk 1]
```diff
@@ -5,10 +5,13 @@ REPO_DIR=$(git rev-parse --show-toplevel)
...
   buildkite/plugin-linter \
     --id lambdatest/lambdatest
 
-popd >/dev/null
\ No newline at end of file
+EXIT_CODE=$?
+
+popd >/dev/null
+
+exit $EXIT_CODE
```
Comment (Delta456) [inline .ci/lint-plugin.sh:17]: Why was this required?

File: .ci/lint-shell.sh [hunk 2]
```diff
@@ -11,4 +11,8 @@ docker run \
   --rm \
   koalaman/shellcheck hooks/*
 
+EXIT_CODE=$?
+
 popd >/dev/null
+
+exit $EXIT_CODE
```
Comment (Delta456) [inline .ci/lint-shell.sh:18]: Same here

File: .ci/lint-plugin.sh (hunk 1 above)
Comment (pearguacamole) [reply .ci/lint-plugin.sh:17]: We are running the tests inside a docker container, which is run from this shell script. So the exit code we get from the tests(inside docker) are lost in the scope of the script. This makes sure that we catch that exit code and exit the shell script with the same exit code so that GitHub actions will know if the tests failed or not. 

If we don't do this, we will get the command line message that the tests failed but the actions will show that the checks have  passed (script ran successfully so exit code 0) even if the tests fail. Hope this answers the question.

File: .ci/lint-shell.sh (hunk 2 above)
Comment (pearguacamole) [reply .ci/lint-shell.sh:18]: Answer is same as before, we do this to catch the exit code of the tests and return them as the exit code of the script, so that GitHub actions knows if the tests failed.

**Comment by navin772** [comment]
Hi @pearguacamole, the changes look good to me, can you please resolve the conflicts as there was a recent push

**Comment by navin772** [comment]
I have resolved the conflicts @pearguacamole

**Review by navin772 — APPROVED** [review]
LGTM

**Comment by navin772** [comment]
@pearguacamole Thanks for the contribution and congratulations on getting the PR merged 🎉 🎉
